### PR TITLE
Support JDK 11 and JDK 17 Compilation

### PR DESCRIPTION
**Pull request by Achintha444**
### Purpose

Update the repository to support JDK 11 and JDK 17 compilation.

### When should this PR be merged

After this PR is merged to the main branch, we can't build the branch on JDK 8

### Version checked,

- openjdk "11.0.16.1"
- openjdk "17.0.4.1"

### Related Issue
https://github.com/wso2/product-is/issues/14224

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -123,28 +123,28 @@
...
         <!--Carbon commons version-->
-        <carbon.commons.version>4.4.8</carbon.commons.version>
+        <carbon.commons.version>4.8.7</carbon.commons.version>
         <carbon.commons.imp.pkg.version>[4.4.0, 5.0.0)</carbon.commons.imp.pkg.version>
         <!--Carbon identity version-->
-        <carbon.identity.framework.version>5.20.34</carbon.identity.framework.version>
+        <carbon.identity.framework.version>5.23.28</carbon.identity.framework.version>
 
-        <carbon.identity.version>5.0.7</carbon.identity.version>
+        <carbon.identity.version>5.23.28</carbon.identity.version>
```
Comment (piraveena) [inline pom.xml:147]: why did you bump the carbon.identity version? Do we have this version

Comment (Achintha444) [reply pom.xml:147]: Yes we have this version.
https://github.com/wso2/carbon-identity-framework/releases/tag/v5.23.28

Comment (piraveena) [reply pom.xml:147]: That is already there as carbon.identity.framework.version

Comment (Achintha444) [reply pom.xml:147]: I will revert it to the 5.0.8, which is the latest version for the carbon.identity repo

**Review by senthalan — APPROVED** [review]

**Review by piraveena — APPROVED** [review]

**Pull request merged**